agent/engine/task_manager_test.go

```diff
@@ -201,8 +201,61 @@ func TestContainerNextState(t *testing.T) {
 			assert.Equal(t, tc.expectedContainerStatus, nextStatus,
 				"Expected next state [%s] != Retrieved next state [%s]",
 				tc.expectedContainerStatus.String(), nextStatus.String())
-			assert.Equal(t, tc.expectedTransitionActionable, actionRequired)
-			assert.Equal(t, tc.expectedTransitionPossible, possible)
+			assert.Equal(t, tc.expectedTransitionActionable, actionRequired, "transition actionable")
+			assert.Equal(t, tc.expectedTransitionPossible, possible, "transition possible")
+		})
+	}
+}
+
+func TestContainerNextStateWithDependencies(t *testing.T) {
+	testCases := []struct {
+		containerCurrentStatus       api.ContainerStatus
+		containerDesiredStatus       api.ContainerStatus
+		dependencyCurrentStatus      api.ContainerStatus
+		expectedContainerStatus      api.ContainerStatus
+		expectedTransitionActionable bool
+		expectedTransitionPossible   bool
```

Comment:
Possible and actionable are always the same in your test cases -- is that coincidence or are there situations where they differ?

Reply:
See here:

https://github.com/samuelkarp/amazon-ecs-agent/blob/0b451f9583ecfcd3959890f03662aa8554dbab8e/agent/engine/task_manager.go#L480-L482

> // 'Stopped, false, true' -> "You can move it to known stopped, but you don't have to call a transition function"
> // 'Running, true, true' -> "You can move it to running and you need to call the transition function"
> // 'None, false, false' -> "This should not be moved; it has unresolved dependencies or is complete; no knownstatus change"

It's not covered in my test cases because I don't need it to be right now.

Reply:
Are those paths covered in other tests? I'm not sure what I can take away from that commented block.

Reply:
Covered in `TestContainerNextState`.